### PR TITLE
Category autocompletion

### LINKED ISSUE #11: Category autocompletion while entering Commons categories
Refinement / specific sub-feature for #3.

Many Wikimedia Commons tools/interfaces allow users to enter / work with Commons categories. Usually, these tools or interfaces offer autocompletion of names of Commons categories. We will make Wikimedia Commons users in OpenRefine quite happy if the Commons Extension also offers this functionality!

Just for inspiration, showing what this looks like in various tools.

image
In the Wikimedia Commons [UploadWizard](https://commons.wikimedia.org/wiki/Special:UploadWizard)

image
In the [HotCat](https://commons.wikimedia.org/wiki/Help:Gadget-HotCat) gadget (note the blue checkmark that appears when the user has selected a correct category name)

image
In the [ISA Tool](https://isa.toolforge.org/campaigns/create) (the user types something without `Category:` but then `Category:` is being displayed)

### PR DESCRIPTION
**Pull request by j-sal**
Fixes #11

### CODE REVIEW

File: module/scripts/index/commons-source-ui.js [hunk 1]
```diff
@@ -18,16 +18,35 @@ Refine.CommonsSourceUI = function(controller) {
 
     var self = this;
 
+    var cmtitle = $.trim(self._elmts.categoryInput[0].value);
+    var inputContainer = $('<div></div>').appendTo(cmtitle);
+    var input = $('<input></input>').appendTo(inputContainer);
+
+    var endpoint = "https://commons.wikimedia.org/w/api.php?action=opensearch&format=json&formatversion=2&search="
+    + cmtitle + "&namespace=14&limit=10";
```
Comment (wetneb) [inline module/scripts/index/commons-source-ui.js:26]: The endpoint should just be the base URL of the API - the user input will be added to the query later on (every time the user input changes)

**Review by wetneb — COMMENTED** [review]
I haven't had the time to look into your `$.suggest.suggest.prototype` problem but have you tried initializing this following the method of `extensions/wikidata/module/scripts/langsuggest.js` instead?

File: module/scripts/index/commons-source-ui.js [hunk 2]
```diff
@@ -18,16 +18,23 @@ Refine.CommonsSourceUI = function(controller) {
...
-    this._elmts.NextButton.click(function(evt) {
-      var cmtitle = $.trim(self._elmts.categoryInput[0].value);
-      if (cmtitle.length === 0) {
-        window.alert($.i18n('commons-source/alert-retrieve'));
-      } else {
-        var doc = {};
-        doc.input = cmtitle;
-        self._controller.startImportingDocument(doc);
-      }
+    var endpoint = "https://commons.wikimedia.org/w/api.php"
```
[the remark this answers is not in the document]
Comment (j-sal) [reply module/scripts/index/commons-source-ui.js:21]: Should this follow the `wikibase-manager.js` implementation?

[the remark this answers is not in the document]
Comment (wetneb) [reply module/scripts/index/commons-source-ui.js:21]: Yes

**Pull request merged**